util/formats/newfmt/types.go

```diff
@@ -0,0 +1,217 @@
+// New Sonar JSON data format (we no longer support any kind of CSV variant).  MetadataObject and
+// ErrorObject are shared between the various data types, everything else is specific to the data
+// type and has a name that clearly indicates that.  The top-level object for all data types is the
+// "...Envelope" object.  Within the envelopes, `Data` and `Errors` are exclusive of each other.
+//
+// Some fields present in the older Sonar data are no longer here, having been deemed redundant or
+// obsolete.  Some are here in a different form.  Therefore, while old and new data are broadly
+// compatible, there may be some minor problems translating between them.
+//
+// The exact semantics of fields are still defined in the spec in doc/DATA-FORMATS.md.
```

Comment:
This file needs to be extracted from the gdoc spec and added.  Actually an argument could be made that it should be a machine-readable spec and that the present file should be generated from that spec.

Reply:
That can't happen until final review however, so we might do it as a separate Issue.